packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -38,10 +38,13 @@ class FlutterDevice {
 
   StreamSubscription<String> _loggingSubscription;
 
-  FlutterDevice(this.device, { bool previewDart2 : false }) {
+  FlutterDevice(this.device, {
+    bool previewDart2 : false,
+    bool trackWidgetCreationLocations : false,
+  }) {
     if (previewDart2)
       generator = new ResidentCompiler(
-        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath));
+        artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath), trackWidgetCreationLocations: trackWidgetCreationLocations);
```

Comment:
[nit] Maybe break this long line.

Reply:
done. Also added curly braces as the statement is getting pretty long to not have them.